src/main/java/org/opensearch/security/securityconf/ConfigModelV6.java

```diff
@@ -542,6 +542,26 @@ public boolean impliesTypePermGlobal(
             roles.stream().forEach(p -> ipatterns.addAll(p.getIpatterns()));
             return ConfigModelV6.impliesTypePerm(ipatterns, resolved, user, actions, resolver, cs);
         }
+
+        @Override
+        public boolean isPermittedOnSystemIndex(String indexName) {
```

Comment:
Do we need this both here and in the newer V7? We reference V6 inside V7 and I don't know how this is planning to be used with the legacy framework.

Reply:
I added `isPermittedOnSystemIndex` to verify system index permission through `SecurityRoles`.
Because I extended the `SecurityRoles` Interface, logic implementation was also required in V6.
https://github.com/opensearch-project/security/pull/4201/files/b3170c2a0a4ba9f1a599b79b7cf66bc66dbeecbf#diff-729d96f84a7b2b9ddced57ed4722f88280fe697df668d9fee80a6979e79742c7R100

Is there another way? Could you explain in more detail, please?